style-guides/javascript/README.md

```diff
@@ -0,0 +1,1479 @@
+# Social Tables JavaScript Style Guide() {
+
+## Table of Contents
+
+  1. [QuickView](#quickview)
+  1. [Types](#types)
+  1. [Objects](#objects)
+  1. [Arrays](#arrays)
+  1. [Strings](#strings)
+  1. [Functions](#functions)
+  1. [Properties](#properties)
+  1. [Variables](#variables)
+  1. [Hoisting](#hoisting)
+  1. [Conditional Expressions & Equality](#conditional-expressions--equality)
+  1. [Blocks](#blocks)
+  1. [Comments](#comments)
+  1. [Whitespace](#whitespace)
+  1. [Commas](#commas)
+  1. [Semicolons](#semicolons)
+  1. [Type Casting & Coercion](#type-casting--coercion)
+  1. [Naming Conventions](#naming-conventions)
+  1. [Accessors](#accessors)
+  1. [Constructors](#constructors)
+  1. [Events](#events)
+  1. [Performance](#performance)
+  1. [Resources](#resources)
+  1. [License](#license)
+
+## Quickview
+  Read this section for main takeaways most relevant to our daily work.
+
+  - **General**
+    + Lines should not be longer than 80 characters
+    + Functions should be < 50 lines **Note: thoughts??**
+    + Semicolons. Use them.
+    + Use camelCase for objects, functions, and instances
+    + Use PascalCase for constructors and classes
+    + Use UPPERCASE for constants
+    + Name your functions. This is helpful for stack traces
+    + Method Chaining: each new method should be on a new line and indented
+    + Never extend built-in prototypes 
+
+  - **Objects**
+
+    + Use the literal syntax for object creation.
+    
+    ```javascript
+    // bad
+    var item = new Object();
+
+    // good
+    var item = {};
+    ```
+  - **Arrays**
+    + Use myArray.push(x) instead of myArray[myArray.length] = x
+    + Use Array#slice to copy an array
+  
+  - **Strings**
+    + **Note: Thoughts on always single quotes vs double quotes? Other say single quotes**
+
+  - **Functions**
+
+  - **Properties**
+    + Access properties via dot notation (myObj.name) rather than subscript notation (myObj[name])
+
+  - **Variables**
+    + Always use `var` to declare variables
+    + Use one `var` declaration per variable
+    + Assign variables at the top of their scope
+    + Declare unassigned variables last
+    + Declare one variable per line **Note: thoughts on this?**
+    + Use names that make sense. Don't use reserved words. Avoid single letters
+
+  - **Hoisting** [Read this section. It's good.](#hoisting)
+
+  - **Conditional Expressions & Equality**
+    + Use `===` and `!==` over `==` and `!=`
+    + Avoid duplication like `if (name !== '')`. Use `if (name)` instead.
+
+  - **Blocks**
+    + Separate into multiple lines and use braces
+      
+      ```javascript
+      // bad
+      if (true) { return true; }
+      
+      // good
+      if (true) {
+        return true;
+      }
+      ```
+
+  - **Comments**
+    + **NOTE: do people prefer `//` or `/** ... */` for multiline comments?**
```

Comment:
If you've already written a one-line comment and need to add to it, you can just start a new line with // instead of changing the first // to a /*

Also I like commenting things out with a simple cmd+/ -- if you know of a keyboard shortcut for for the other way then I cede this point